docs/resources/autoscaling_group.md

```diff
@@ -0,0 +1,136 @@
+      size           = 10
+      type           = "HDD"
+      user_data      = "ZWNobyAiSGVsbG8sIFdvcmxkIgo="
+      image_password = random_password.server_image_password.result
+      boot_order     = "AUTO"
+    }
+  }
+}
+
+resource "random_password" "server_image_password" {
+  length           = 16
+  special          = false
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+- `max_replica_count` - (Required)[int] The maximum value for the number of replicas on a VM Auto Scaling Group. Must be >= 0 and <= 200. Will be enforced for both automatic and manual changes.
+- `min_replica_count` - (Required)[int] The minimum value for the number of replicas on a VM Auto Scaling Group. Must be >= 0 and <= 200. Will be enforced for both automatic and manual changes.
+- `target_replica_count` - (Optional)[int] The target number of VMs in this Group. Depending on the scaling policy, this number will be adjusted automatically. VMs will be created or destroyed automatically in order to adjust the actual number of VMs to this number. If targetReplicaCount is given in the request body then it must be >= minReplicaCount and <= maxReplicaCount.
+- `name` - (Required)[string] User-defined name for the Autoscaling Group.
+- `policy` - (Required) Specifies the behavior of this Autoscaling Group. A policy consists of Triggers and Actions, whereby an Action is some kind of automated behavior, and a Trigger is defined by the circumstances under which the Action is triggered. Currently, two separate Actions, namely Scaling In and Out are supported, triggered through Thresholds defined on a given Metric.
+    - `metric` - (Required)[string] The Metric that should trigger the scaling actions. Metric values are checked at fixed intervals. Possible values: `INSTANCE_CPU_UTILIZATION_AVERAGE`, `INSTANCE_NETWORK_IN_BYTES`, `INSTANCE_NETWORK_IN_PACKETS`, `INSTANCE_NETWORK_OUT_BYTES`, `INSTANCE_NETWORK_OUT_PACKETS`
+    - `range` - (Optional)[string] Defines the time range, for which the samples will be aggregated. Default is 120s. *Note that when you set it to values like 5m the API will automatically transform it in PT5M, so the plan will show you a diff in state that should be ignored.*
+    - `scale_in_action` - (Required) Specifies the action to take when the `scaleInThreshold` is exceeded. Hereby, scaling in is always about removing VMs that are currently associated with this autoscaling group. Default termination policy is OLDEST_SERVER_FIRST.
+        - `amount` - (Required)[int] When `amountType == ABSOLUTE`, this is the number of VMs added or removed in one step. When `amountType == PERCENTAGE`, this is a percentage value, which will be applied to the autoscaling group's current `targetReplicaCount` in order to derive the number of VMs that will be added or removed in one step. There will always be at least one VM added or removed. For SCALE_IN operation now volumes are NOT deleted after the server deletion.
```

Comment:
Since this is the amount of the scale_in_action, which always performs removals it should state:
"There will always be at least one VM removed when specifying `PERCENTAGE` amountType" or something similar.
Also, the volume part is a bit unclear, does it mean that volumes attached to the VMs selected for removal, are not removed ?

Reply:
@cbeti-ionos please read the new description, for me it is still unclear.

Reply:
the volume removal behavior for the scale_in_action is explained by the description of the `delete_volumes` attribute, but it's currently missing from the docs